pytorch_lightning/core/lightning.py

```diff
@@ -1921,31 +1923,32 @@ def to_onnx(
         self,
         file_path: Union[str, Path],
         input_sample: Optional[Any] = None,
+        model_check_fn: Callable = None,
         **kwargs,
-    ):
+    ) -> None:
         """
         Saves the model in ONNX format.
 
         Args:
             file_path: The path of the file the onnx model should be saved to.
             input_sample: An input for tracing. Default: None (Use self.example_input_array)
+            model_check_fn: custom function that conducts onnx model checks.
             **kwargs: Will be passed to torch.onnx.export function.
 
-        Example:
-            >>> class SimpleModel(LightningModule):
-            ...     def __init__(self):
-            ...         super().__init__()
-            ...         self.l1 = torch.nn.Linear(in_features=64, out_features=4)
-            ...
-            ...     def forward(self, x):
-            ...         return torch.relu(self.l1(x.view(x.size(0), -1)))
+        Example::
 
-            >>> with tempfile.NamedTemporaryFile(suffix='.onnx', delete=False) as tmpfile:
-            ...     model = SimpleModel()
-            ...     input_sample = torch.randn((1, 64))
-            ...     model.to_onnx(tmpfile.name, input_sample, export_params=True)
-            ...     os.path.isfile(tmpfile.name)
-            True
+            class SimpleModel(LightningModule):
```

Comment:
reviewer: see https://github.com/PyTorchLightning/pytorch-lightning/pull/7458#issuecomment-875989784
hard to know what's going on
doctest environment is somehow different. easiest to fix is to convert the doctest to a regular example. 
we have the normal tests in our test suite